### PR TITLE
init rolling file appender

### PR DESCRIPTION
**Pull request by urso**

### CODE REVIEW

File: backend/appender/rolling/rolling.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+	mu sync.Mutex
+	wg sync.WaitGroup
+
+	file *os.File
+	stat FileInfo
+}
+
+// Rolloverer is used by the trigger to start the rollover process.
+type Rolloverer interface {
```
Comment (michalpristas) [inline backend/appender/rolling/rolling.go]: can we come up with a better name. I think `-er` rule is not always the best option. I would settle with simple `Roller`, `LogRoller`, `FileRoller` or `FileRotator`
This is just a small nit but i think it slightly improves readability especially with exported interfaces

Comment (urso) [reply backend/appender/rolling/rolling.go]: I renamed it to 'Rotator'.

**Comment by urso** [comment]
@michalpristas 	@ph The rolling file appender is basically working (the whole project needs some tests :) ). Only thing I wonder is: Do we want to be able to configure a limit on log message sizes (bigger ones will be filtered out)? If so, how to configure it? Right now the file size is configured via 'trigger' (which checks the size after writing the log message).

I include the data in the rotated names, this reduces the number of rename operations we have to run, removes the file deletion problems with external collectors (like filebeat) on windows.

**Comment by ph** [comment]
> @michalpristas @ph The rolling file appender is basically working (the whole project needs some tests :) ). Only thing I wonder is: Do we want to be able to configure a limit on log message sizes (bigger ones will be filtered out)? If so, how to configure it? Right now the file size is configured via 'trigger' (which checks the size after writing the log message).

Lets get this merge and we can create another issue.

Sorry I've missed the notification somehow.

**Review by ph — APPROVED** [review]

File: backend/appender/rolling/appender.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+func (a *Appender) FileStat() FileInfo {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	return a.stat
+}
+
+func (a *Appender) Write(b []byte) (int, error) {
+	if a.file == nil {
```
Comment (michalpristas) [inline backend/appender/rolling/appender.go:207]: exported method, is it safe to omit lock here? this possibly calls rotate

**Review by michalpristas — APPROVED** [review]

File: backend/appender/rolling/appender.go (hunk 2 above)
Comment (urso) [reply backend/appender/rolling/appender.go:207]: The writer is not called directly from the process, but it is called from the 'layout'. That is the lock is already taken. Will try to make this method 'unexported'.

Comment (urso) [reply backend/appender/rolling/appender.go:207]: Did update the code such that Write is not exported anymore (plus added some comments).

**Comment by ph** [comment]
Thanks @urso

**Pull request merged**